src/common/k8s.ts

```diff
@@ -59,13 +59,13 @@ export interface DeploymentState {
 }
 
 export const getDeploymentState = async (): Promise<DeploymentState> => {
-  if (env.DISABLE_SYNC) return {}
+  if (env.isDev && env.DISABLE_SYNC) return {}
```

Comment:
Everywhere in code I see a pair `env.isDev && env.DISABLE_SYNC`

I think DISABLE_SYNC should always set to false if `env.isDev=false`, then you can use a very simple if statement through the code.
Also if DISABLE_SYNC is meant to use only in dev mode them perhaps it should have DEV_ prefix, i.e.: DEV_DISABLE_SYNC

Reply:
no, those are separately used

Reply:
if you care to rename env vars in another PR, be my guest...

Reply:
Good idea, but it is a bit of a stint to make envalid deal with logic. Let's keep tha for another time as it's not changed in this PR